assets/js/RenameColumns.test.js

```diff
@@ -0,0 +1,240 @@
+                            },
+                            value: JSON.stringify({
+                                'name': 'host_name',
+                                'narrative': 'nrtv'
+                            })
+                        }
+                    ]
+                }
+            ]
+        }
+    };
+
+    const addModuleResponse = {
+        id: NEW_RENAME_WFM_ID,
+        module_version: {
+            module: {
+                id_name: 'rename-columns'
+            }
+        },
+        parameter_vals: [
+            {
+                id: NEW_RENAME_DISPLAY_ALL_PAR_ID,
+                parameter_spec: {
+                    id_name: 'display-all'
+                },
+                // value defaults to true when just added
+                value: true
+            },
+            {
+                id: NEW_RENAME_ENTRIES_ID,
+                parameter_spec: {
+                    id_name: 'rename-entries'
+                },
+                value: ''
+            }
+        ]
+    };
+
+    beforeEach(() => {
+        api = {
+            onParamChanged: jest.fn().mockReturnValue(Promise.resolve()),
+            addModule: jsonResponseMock(addModuleResponse),
+            setSelectedWfModule: jest.fn().mockReturnValue(Promise.resolve())
+        };
+        mockAPI(api);
+        mockStoreAPI(api);
+        store = {
+            getState: () => initialState,
+            dispatch: jest.fn().mockReturnValue(Promise.resolve())
+        };
+        mockStore(store);
+    });
+
+    it('Adds a new rename module after the current non-rename module '
+        + 'if a rename module does not after it and sets parameters correctly', (done) => {
+        updateRename(LOADURL_WFM_ID, {
+            prevName: 'cornerstone',
+            newName: 'cs'
+        });
+        // The entries that should be sent to server is {'cornerstone': 'cs'}
+        setImmediate(() => {
+            // Checks the module adding part
+            expect(api.addModule.mock.calls).toHaveLength(1);
+            // The id of the added module should be renameModuleId
+            expect(api.addModule.mock.calls[0][1]).toBe(initialState.renameModuleId);
+            // The index of the added module should be 1 as we are inserting after the 0th module (loadurl)
+            expect(api.addModule.mock.calls[0][2]).toBe(1);
+
+            // Checks the parameter setting part, which should have two parts:
+            // 1. Setting loadAll to false
+            // 2. Setting the new rename entries
+            expect(api.onParamChanged.mock.calls).toHaveLength(2);
+            // Check that we set loadAll to false
+            expect(api.onParamChanged.mock.calls[0][0]).toBe(NEW_RENAME_DISPLAY_ALL_PAR_ID);
+            expect(api.onParamChanged.mock.calls[0][1].value).toBe(false);
+            // Check that we set the new entries
+            expect(api.onParamChanged.mock.calls[1][0]).toBe(NEW_RENAME_ENTRIES_ID);
```

Comment:
These two lines can replace _all_ ther other `expect` calls on `api.onParamChanged`.
```javascript
expect(api.onParamChanged).toHaveBeenNthCalledWith(1, NEW_RENAME_DISPLAY_ALL_PAR_ID, false);
expect(api.onParamChanged).toHaveBeenNthCalledWith(2, NEW_RENAME_ENTRIES_ID, [ { cornerstone: 'cs' } ]);
```

Ditto above: `expect(api.addModule).toHaveBeenCalledWith([fill me in], initialState.renameModuleId, 1)`